### PR TITLE
Adding SQL Standard Changes for BMP

### PR DESCRIPTION
**Pull request by BIHxSALKO**
In this pull request, I am presenting changes to the base SQL Style Guide to conform to the standards discussed. My edits pertain to stored procedure creation syntax, in particular, how we treat in/out params and local vars, indentation on JOINs, and indentations on AND/OR keywords after the first WHERE declaration. Additionally, semicolons in queries to further indicate that this is for a MySQL dev environment.

### CODE REVIEW

File: _includes/sqlstyle.guide.md [hunk 1]
```diff
@@ -106,6 +105,32 @@ FROM staff AS s;
 * The name must contain a verb.
 * Do not prefix with `sp_` or any other such descriptive prefix or Hungarian
   notation.
+* Prefix arguments/variables with an underscore.
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: How about adding a rationale here? Something like: "This is done to visually differentiate variables from other identifiers such as column and table names."

File: _includes/sqlstyle.guide.md [hunk 2]
```diff
@@ -106,6 +105,32 @@ FROM staff AS s;
...
+* Align argument types with longest argument name + 1 space.
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: How about adding "in order to create a separate visual column for the type declarations." ?

File: _includes/sqlstyle.guide.md [hunk 3]
```diff
@@ -106,6 +105,32 @@ FROM staff AS s;
...
+* All IN parameters should be listed before OUT parameters.
+* Arguments/variables meant to be immutable are screaming snake caps.
+* Arguments/variables meant to be mutable are lowercase snake caps.
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: I would use the term "[snake case](https://en.wikipedia.org/wiki/Snake_case)" over "snake caps" as I think it is more common. Also instead of "are" how about "should be in"?

File: _includes/sqlstyle.guide.md [hunk 4]
```diff
@@ -106,6 +105,32 @@ FROM staff AS s;
...
+  _lkp_id_id2        INT,  -- you can change this
+  OUT _something_id  INT,
+  OUT _something_id2 INT,
+  OUT _something_id3 INT
+)
+
+BEGIN
+
+DECLARE _SOMETHING_ID VARCHAR(255) DEFAULT '123'; -- allowed
+SET _SOMETHING_ID = '123'; -- not allowed
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: I would probably change this example to set the value to something else to drive the point home that the value can not change.

File: _includes/sqlstyle.guide.md [hunk 5]
```diff
@@ -106,6 +105,32 @@ FROM staff AS s;
...
+* All IN parameters should be listed before OUT parameters.
+* Arguments/variables meant to be immutable are screaming snake caps.
+* Arguments/variables meant to be mutable are lowercase snake caps.
+* Default to immutable naming conventions.
+
+```sql
+CREATE PROCEDURE `test`(
+  IN
+  _LKP_ID            INT,  -- not supposed to change this, even though you can!
+  _lkp_id_id2        INT,  -- you can change this
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: I would probably add ", but changing input values is discouraged" to the comment.

File: _includes/sqlstyle.guide.md [hunk 6]
```diff
@@ -217,8 +242,8 @@ Always include newlines/vertical space:
 * to separate code into related sections, which helps to ease the readability of
   large chunks of code.
 
-Keeping all the dependent keywords (e.g. `ON` depends on `JOIN`, `AND` depends on `WHERE`,
-`SET` depends on `UPDATE` right aligned with the top level keywords helps make it clear
+Keeping the top most dependent keywords (e.g. `ON` depends on `JOIN`, `AND` depends on `WHERE`,
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: I am honestly a little confused by what "top most" keywords means here. What would a non-top most keyword be?

File: _includes/sqlstyle.guide.md [hunk 7]
```diff
@@ -229,7 +254,7 @@ VALUES ('Charcoal Lane', '1990-01-01 01:01:01.00000', '1990-01-01 01:01:01.00000
 
 ```sql
 UPDATE albums
-   SET release_date = '1990-01-01 01:01:01.00000'
+  SET release_date = '1990-01-01 01:01:01.00000'
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: I think that by our rules, the OR in the following SELECT example should only be indented two spaces.

File: _includes/sqlstyle.guide.md [hunk 8]
```diff
@@ -246,27 +271,34 @@ WHERE a.title = 'Charcoal Lane'
 
 #### JOINs
 
-Joins should be indented 2 spaces right from the `FROM` keyword
+Joins should be left defined relative to the `FROM` keyword
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md]: "left defined relative to" is confusing. How about just: "Joins should be directly underneath the `FROM` keyword with no indentation."?

File: _site/CNAME [hunk 9]
```diff
@@ -0,0 +1 @@
+www.my.sqlstyle.guide
```
Comment (gerryster) [inline _site/CNAME]: Is this file used by the static site generator? If so, how about adding it to .gitignore?

File: _includes/sqlstyle.guide.md (hunk 1 above)
Comment (BIHxSALKO) [reply _includes/sqlstyle.guide.md]: Added supporting language as recommended.

**Comment by BIHxSALKO** [comment]
Closing pull request, and re-submitting with suggested changes and adding in two requested changes.

File: _includes/sqlstyle.guide.md [hunk 10]
```diff
@@ -106,6 +105,39 @@ FROM staff AS s;
...
+  _lkp_id_id2        INT,  -- you can change this  but changing input values is discouraged
+  OUT _something_id  INT,
+  OUT _something_id2 INT,
+  OUT _something_id3 INT
+)
+
+BEGIN
+
+DECLARE _CNST_HIDDEN_COMPONENT VARCHAR(255) DEFAULT 'Hidden Component'; -- allowed
+SET _CNST_HIDDEN_COMPONENT = 'Hidden Component'; -- not allowed
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md:131]: How about changing this line to the following to make it very clear that the value is changing?

```sql
SET _CNST_HIDDEN_COMPONENT = 'CHANGED Component'; -- not allowed
```

File: _includes/sqlstyle.guide.md [hunk 11]
```diff
@@ -229,44 +270,55 @@ VALUES ('Charcoal Lane', '1990-01-01 01:01:01.00000', '1990-01-01 01:01:01.00000
...
+  AND artists.group_count > 2
 WHERE a.title = 'Charcoal Lane'
-   OR a.title = 'The New Danger';
+  OR a.title = 'The New Danger';
 ```
 
 #### JOINs
 
-Joins should be indented 2 spaces right from the `FROM` keyword
+Joins should be directly underneath the FROM keyword with no indentation.
```
Comment (gerryster) [inline _includes/sqlstyle.guide.md:294]: This is a bit pedantic but how about changing the end to "with no additional indentation"? The entire query may be indented as part of the proc definition.

File: _includes/sqlstyle.guide.md [hunk 12]
```diff
@@ -416,11 +468,22 @@ SELECT
...
 FROM office_locations
 WHERE country = 'United Kingdom'
   AND opening_time BETWEEN 8 AND 9
-  AND postcode IN ('EH1', 'BN1', 'NN1', 'KW1')
+  AND postcode IN ('EH1', 'BN1', 'NN1', 'KW1');
+```
+
+* Use `TRUE` and `FALSE` instead of `0` and `1` as that it is immediately
+  obvious that these values are booleans and not magic numbers. This is also
+  more semantic. It appears as though SQL99 introduces this "IS FALSE" or "IS
```
[the remark this answers is not in the document]
Comment (gerryster) [reply _includes/sqlstyle.guide.md]: Don't forget to remove this.

File: .gitignore [hunk 13]
```diff
@@ -7,6 +7,7 @@ nbproject
 ._*
 Thumbs.db
 Gemfile.lock
+_site/CNAME
```
Comment (gerryster) [inline .gitignore:10]: I /think/ this is redundant as the entire "_site" directory should be ignored per line 12. Also the _site/CNAME file is still on your branch and should most likely be removed.

**Comment by gerryster** [comment]
@BIHxSALKO, can you make the last couple of updates so we can close this out?

File: _includes/sqlstyle.guide.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (BIHxSALKO) [reply _includes/sqlstyle.guide.md]: @gerryster , I do not see a reference any longer to SQL99. In my branch, you can view the styleguide [here](https://github.com/BIHxSALKO/sqlstyle.guide/blob/gh-pages/_includes/sqlstyle.guide.md), do a cntrl-f for 'SQL99' and there isn't a reference to that statement containing "SQL99".
In this commit https://github.com/bluemarblepayroll/sqlstyle.guide/pull/11/commits/78bc188a1bf61a3535de9075d50bdc416dda1ffc
I remove the reference in lines 480-481.

File: CNAME [hunk 14]
```diff
@@ -1 +1 @@
-www.sqlstyle.guide
+www.mysqlstyle.guide
```
Comment (gerryster) [inline CNAME:1]: Why was this changed?

File: _site/CNAME [hunk 15]
```diff
@@ -1 +1 @@
-www.mysqlstyle.guide
+www.sqlstyle.guide
```
Comment (gerryster) [inline _site/CNAME:1]: This entire file should not exist in version control. Note that in https://github.com/bluemarblepayroll/sqlstyle.guide, the _site directory does not exist. I think this only used by the static site generator. Think of this as a directory to store compiled assets which should not be checked into version control.

File: _includes/sqlstyle.guide.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (gerryster) [reply _includes/sqlstyle.guide.md]: Ok, thanks.

**Comment by gerryster** [comment]
@BIHxSALKO : there is still some weirdness with the CNAME file. I am seeing this in the diff when I am only expecting to see the Markdown file changed:

image

Can you address this please?
